### PR TITLE
Export: Single panel export via panel menu

### PR DESCRIPTION
**Pull request by sjd210**
**What is this feature?**

Adds support for quick and easy image/data export via the panel menu. This can be accessed through the top-right menu icon of a panel, then hovering over the "Export" element. The following formats are currently supported:

**IMAGE:** PNG, JPEG, BMP

image

**DATA:** CSV, Excel, Data JSON, DataFrame JSON, Panel JSON (as from the Inspect menu).

image

Sample files of each format are accessible here: [Panel Export Samples.zip](https://github.com/grafana/grafana/files/12215349/Panel.Export.Samples.zip)

**Why do we need this feature?**

Previously, the only way to access data exports from a panel was in the Inspect menu. This is slow, doesn't allow for true XLSX formatted data (instead a CSV file with a header, so any changes to the format of the file once downloaded wouldn't be maintained), and additionally required manual copy and pasting text from the JSON menu as the only way to access JSON data.

This feature allows for data to be downloaded in any of these formats in one click from the panel menu - making it easier to find and quicker to use, as well as supporting image formats download (which were inaccesible prior).

**Who is this feature for?**

All users, particularly those with need to use or display data from Grafana externally.

**Special notes for your reviewer:**

Please check that:
- [x] It works as expected from a user's perspective.
- [x] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc. **(Need to do)**

### CODE REVIEW

File: public/app/features/dashboard/utils/getPanelMenu.ts [hunk 1]
```diff
@@ -147,6 +164,90 @@ export function getPanelMenu(
...
+    exportMenu.push({
+      text: t('panel.header-menu.export-json', `Panel JSON`),
+      iconClassName: 'book',
+      onClick: () => onExportPanel(ExportType.panelJson),
+    });
+  }
+
+  // TODO: Re-implement real feature toggle
+
+  const featureToggleEnabled = true;
```
Comment (mdvictor) [inline public/app/features/dashboard/utils/getPanelMenu.ts:240]: Looks like this still needs to be implemented, I think

**Comment by ryantxu** [comment]
This looks good!  but I still want to split out the UI/UX part from the functional parts.  The duplicate/overlap with the inspector feel a bit off to me -- I am inclined to fix+improve what we have rather than building a parallel/similar track next to it.

Also I am not sure (genuinely) if this should be on the top level menu... perspective is hard since it so depends on user case.

**Comment by sjd210** [comment]
@ryantxu What are your thoughts on lifting the feature to be part of the `Inspect` panel - adding a new subpanel (alongside Data/Stats/JSON/Query) either for Images specifically or for Exports generally as in the current PR? 

Alternatively, to address this being in the top-level menu we could add it to `More...` or even to `Inspect` directly in the panel menu (without having to click through) to maintain the quick accessibility, if we want to conserve that?

**Review by mckn — COMMENTED** [review]
Code looks good! I'm not sure about the UI pattern being used here. I would make the menu a lot simpler. Maybe even only have "Export" that would open a modal where you can select if you would like to export the panel as an image or download its data.

Depending on your choice I would then prompt for image- or data format.

File: public/app/features/dashboard/utils/getPanelMenu.ts (hunk 1 above)
Comment (sjd210) [reply public/app/features/dashboard/utils/getPanelMenu.ts]: Got that sorted now 👍

**Comment by sjd210** [comment]
I'll have a discussion with the Saga and UX teams about what might be the best option to address the UI concerns here and then get back to this. Any more feedback in the meantime is still appreciated.

**Comment by mdvictor** [comment]
@ryantxu I agree we shouldn't duplicate functionality, but I'm also not sure that the entire export should sit in the inspector. My thinking is that this should be a quality of life improvement that makes it easy for a user to download their panels in text or image formats, while the inspector is more of a debugging tool that allows you to look at/export your data either formatted or raw.

I would personally remove the JSON exports since they are more relevant in the inspector context, and add other more relevant data export types, for example [custom format exports](https://github.com/grafana/grafana/issues/25138), maybe in another PR.

**Comment by GrafanaWriter** [comment]
I'll remove the type/docs label as I don't see any updates to the documentation. Please ping the team when you're ready for docs review

**Pull request closed without merging**